packages/beacon-node/src/chain/chain.ts

```diff
@@ -869,6 +873,24 @@ export class BeaconChain implements IBeaconChain {
     if (headState) {
       this.opPool.pruneAll(headState, finalizedState);
     }
+
+    // TODO: finalizedState may not be available after https://github.com/ChainSafe/lodestar/issues/5968.
+    // In that case implement getCheckpointStateValidatorCount() in regen and use headState instead
+    if (finalizedState?.epochCtx.isAfterEIP6110()) {
```

Comment:
I would prefer for `finalizedState == falsy` to be handled explicitly, the `?.` is deceptive on its implications